simq/src/main/java/com/simprints/simq/utils/QualityUtils.kt

```diff
@@ -0,0 +1,100 @@
+package com.simprints.simq.utils
+
+import android.graphics.Bitmap
+import kotlin.math.exp
+import kotlin.math.roundToInt
+import kotlin.math.sqrt
+
+object QualityUtils {
+    
+    /**
+     * Ramp scoring function: linear interpolation between min and max.
+     * 
+     * @param x The input value to score
+     * @param min Minimum threshold (scores 0.0 below this)
+     * @param max Maximum threshold (scores 1.0 above this)
+     * @return Score between 0.0 and 1.0
+     */
+    fun rampScore(x: Double, min: Double, max: Double): Double {
+        return when {
+            x < min -> 0.0
+            x > max -> 1.0
+            else -> (x - min) / (max - min)
+        }
+    }
+    
+    /**
+     * Plateau scoring function: optimal range with smooth sigmoid falloff.
+     * 
+     * @param x The input value to score
+     * @param centerLow Lower bound of optimal range
+     * @param centerHigh Upper bound of optimal range
+     * @param edgeLow Lower edge threshold
+     * @param edgeHigh Upper edge threshold
+     * @param steepness Steepness of sigmoid falloff
+     * @return Score between 0.0 and 1.0
+     */
+    fun plateauScore(
```

Comment:
Catch-all "Utils" are usually a code smell. I would suggest adding `ScoringFunctions` class/object and `BitmapExt.kt` with top-level extension functions for bitmap manipulations.